### PR TITLE
Go2rtc 1.9.4

### PR DESCRIPTION
**Pull request by skrashevich**
## Description

Update go2rtc to version 1.9.4

## Checklist

- [x] Build rule `all-supported` completed successfully
- [x] New installation of package completed successfully
- [x] Package upgrade completed successfully (Manually install the package again)
- [x] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Package-Update-Policy#tests-checks)
- [x] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created

### Type of change

- [ ] Bug fix
- [ ] New Package
- [x] Package update
- [ ] Includes small framework changes
- [ ] This change requires a documentation update (e.g. Wiki)

Obsolete #6086 and #6093

### CODE REVIEW

**Comment by hgy59** [comment]
@skrashevich version 1.9.3 is avalable...

File: spk/go2rtc/Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 SPK_NAME = go2rtc
-SPK_VERS = 1.8.5
-SPK_REV = 4
+SPK_VERS = 1.9.2
+SPK_REV = 8
```
Comment (hgy59) [inline spk/go2rtc/Makefile:3]: it is not required to increment SPK_REV within a PR. So `SPK_REV = 5` is expected here.

File: spk/go2rtc/Makefile [hunk 2]
```diff
@@ -10,7 +10,7 @@ UNSUPPORTED_ARCHS = $(PPC_ARCHS)
 MAINTAINER = skrashevich
 DESCRIPTION = Ultimate camera streaming application with support RTSP, RTMP, HTTP-FLV, WebRTC, MSE, HLS, MP4, MJPEG, HomeKit, FFmpeg, etc
 DISPLAY_NAME = go2rtc
-CHANGELOG = "Enhanced Rockchip transcoding, OPUS codec support, WebUI log viewer, Tapo and HomeKit source handling. Improved H265, FFmpeg, CORS support. Fixed D-Link SDP, exec source issues, H265 hardware profiles, OPUS audio, mdns log, Aqara G2H compatibility, reconnect panic."
+CHANGELOG = "The latest update includes various improvements and bug fixes such as enhanced ASCII streaming support, WebRTC loading optimizations, and fixes for HomeKit configuration issues."
```
Comment (hgy59) [inline spk/go2rtc/Makefile:13]: We are used to document changes related to the synology package only, and not related to the upsteam application.
If you want, you can provide link(s) to upstream change log(s) (see spk/bazarr/Makefile for an example).

**Comment by skrashevich** [comment]
> @skrashevich version 1.9.3 is avalable...

will wait for 1.9.4? 😆

**Comment by hgy59** [comment]
> will wait for 1.9.4? 😆

wait for release with https://github.com/AlexxIT/go2rtc/issues/1172

**Comment by skrashevich** [comment]
> > will wait for 1.9.4? 😆
> 
> wait for release with [AlexxIT/go2rtc#1172](https://github.com/AlexxIT/go2rtc/issues/1172)

Perhaps i'll add a patch to this PR to fix the https://github.com/AlexxIT/go2rtc/issues/1172 issue instead of waiting for a new release?

```patch
From 0667683e4d6b25dd17d5e182f255b71989e4e133 Mon Sep 17 00:00:00 2001
From: Alex X <alexey.khit@gmail.com>
Date: Fri, 7 Jun 2024 17:57:36 +0300
Subject: [PATCH] Restore support old cipher suites after go1.22 #1172

---
 pkg/tcp/request.go | 20 ++++++++++++++++++--
 1 file changed, 18 insertions(+), 2 deletions(-)

diff --git a/pkg/tcp/request.go b/pkg/tcp/request.go
index 88da83b6..13463cd7 100644
--- a/pkg/tcp/request.go
+++ b/pkg/tcp/request.go
@@ -19,11 +19,11 @@ func Do(req *http.Request) (*http.Response, error) {
 
 	switch req.URL.Scheme {
 	case "httpx":
-		secure = &tls.Config{InsecureSkipVerify: true}
+		secure = insecureConfig
 		req.URL.Scheme = "https"
 	case "https":
 		if hostname := req.URL.Hostname(); IsIP(hostname) {
-			secure = &tls.Config{InsecureSkipVerify: true}
+			secure = insecureConfig
 		}
 	}
 
@@ -144,6 +144,22 @@ type key string
 var connKey = key("conn")
 var secureKey = key("secure")
 
+var insecureConfig = &tls.Config{
+	InsecureSkipVerify: true,
+	CipherSuites: []uint16{
+		tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256, tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+		tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+		tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305, tls.TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,
+		tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA, tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
+		tls.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA, tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
+
+		// this cipher suites disabled starting from https://tip.golang.org/doc/go1.22
+		// but cameras can't work without them https://github.com/AlexxIT/go2rtc/issues/1172
+		tls.TLS_RSA_WITH_AES_128_GCM_SHA256, // insecure
+		tls.TLS_RSA_WITH_AES_256_GCM_SHA384, // insecure
+	},
+}
+
 func WithConn() (context.Context, *net.Conn) {
 	pconn := new(net.Conn)
 	return context.WithValue(context.Background(), connKey, pconn), pconn
 ```

File: cross/go2rtc/Makefile [hunk 3]
```diff
@@ -27,4 +27,4 @@ include ../../mk/spksrc.cross-go.mk
 go2rtc_post_extract:
 	@$(MSG) Rename extracted folder to make imports work when used as GOPATH
 	cd $(EXTRACT_PATH) && [ -d "$(PKG_NAME)-$(PKG_VERS)" ] && mv "$(PKG_NAME)-$(PKG_VERS)" $(PKG_NAME)
-	cd "$(EXTRACT_PATH)/$(PKG_NAME)" && env $(ENV) go get github.com/tillberg/autorestart
+	cd "$(EXTRACT_PATH)/$(PKG_NAME)"
```
Comment (hgy59) [inline cross/go2rtc/Makefile:30]: This line is obsolete

```suggestion
```

### Proposal:

With the following changes, there is no need to rename the folder. 

Without POST_EXTRACT_TARGET and one level of sub folder removed, the source dir will be `$(WORK_DIR)/src/github.com/$(PKG_NAME)-$(PKG_VERS)`:


```diff
diff --git "a/cross/go2rtc/Makefile" "b/cross/go2rtc/Makefile"
index 58b25d8b..1b33a42c 100644
--- "a/cross/go2rtc/Makefile"
+++ "b/cross/go2rtc/Makefile"
@@ -4,8 +4,7 @@ PKG_EXT = tar.gz
 PKG_DIST_NAME = v$(PKG_VERS).$(PKG_EXT)
 PKG_DIST_SITE = https://github.com/AlexxIT/go2rtc/archive
 PKG_DIST_FILE = $(PKG_NAME)-$(PKG_VERS).$(PKG_EXT)
-EXTRACT_PATH = $(WORK_DIR)/src/github.com/$(PKG_NAME)
-PKG_DIR = src/github.com/$(PKG_NAME)/$(PKG_NAME)
+EXTRACT_PATH = $(WORK_DIR)/src/github.com/
 
 BUILD_DEPENDS = native/go
 
@@ -13,18 +12,10 @@ HOMEPAGE = https://github.com/AlexxIT/go2rtc/
 COMMENT  = Ultimate camera streaming application with support RTSP, RTMP, HTTP-FLV, WebRTC, MSE, HLS, MP4, MJPEG, HomeKit, FFmpeg, etc.
 LICENSE  = MIT
 
-POST_EXTRACT_TARGET = go2rtc_post_extract
-
-GO_SRC_DIR = $(EXTRACT_PATH)/$(PKG_NAME)
+GO_SRC_DIR = $(EXTRACT_PATH)/$(PKG_NAME)-$(PKG_VERS)
 GO_BIN_DIR = $(GO_SRC_DIR)/$(PKG_NAME)
 
 GO_BUILD_ARGS = -trimpath
 GO_LDFLAGS = -s -w
 
 include ../../mk/spksrc.cross-go.mk
-
-.PHONY:  go2rtc_post_extract
-go2rtc_post_extract:
-	@$(MSG) Rename extracted folder to make imports work when used as GOPATH
-	cd $(EXTRACT_PATH) && [ -d "$(PKG_NAME)-$(PKG_VERS)" ] && mv "$(PKG_NAME)-$(PKG_VERS)" $(PKG_NAME)
-	cd "$(EXTRACT_PATH)/$(PKG_NAME)"

```


just ensure the path of the license file must be adjusted in spk/go2rtc/Makefile:
```diff
diff --git "a/spk/go2rtc/Makefile" "b/spk/go2rtc/Makefile"
index 825fc8e9..3d7182d2 100644
--- "a/spk/go2rtc/Makefile"
+++ "b/spk/go2rtc/Makefile"
@@ -11,7 +11,7 @@ MAINTAINER = skrashevich
 DESCRIPTION = Ultimate camera streaming application with support RTSP, RTMP, HTTP-FLV, WebRTC, MSE, HLS, MP4, MJPEG, HomeKit, FFmpeg, etc
 DISPLAY_NAME = go2rtc
 CHANGELOG = "go2rtc updated to v1.9.4"
-LICENSE_FILE = $(WORK_DIR)/src/github.com/go2rtc/go2rtc/LICENSE
+LICENSE_FILE = $(WORK_DIR)/src/github.com/go2rtc-$(SPK_VERS)/LICENSE
 HOMEPAGE = https://github.com/AlexxIT/go2rtc
 LICENSE = MIT
 ```

**Review by hgy59 — APPROVED** [review]
Is this version worth to publish?
There are still some open PRs upstream

**Comment by skrashevich** [comment]
Open pull requests exist in any active project :)
In my opinion, version 1.9.4 is quite stable for production use.

But we can always ask @AlexxIT's opinion ;)

**Comment by hgy59** [comment]
@skrashevich is it not normal that the new button "Net" => network.html displays an empty page (I have not configured any cam or stream)?

**Comment by AlexxIT** [comment]
It only shows "active" data.

**Comment by hgy59** [comment]
> It only shows "active" data. https://www.youtube.com/watch?v=ZzePTFl0QFk

Thanks, a cool feature 🚀

**Pull request merged**